libbpf-tools/sigsnoop.bpf.c

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause)
+/* Copyright (c) 2021~2022 Hengqi Chen */
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include "sigsnoop.h"
+
+#define MAX_ENTRIES	10240
+
+const volatile pid_t target_pid = 0;
+const volatile int target_signal = 0;
+const volatile bool failed_only = false;
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_ENTRIES);
+	__type(key, __u32);
+	__type(value, struct event);
+} values SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
+	__uint(key_size, sizeof(__u32));
+	__uint(value_size, sizeof(__u32));
+} events SEC(".maps");
+
+static int probe_entry(pid_t target_pid, int sig)
```

Comment:
Hi.

I did not realize it during my first review but the function argument `target_pid` hides the global `target_pid`, so PID filtering will be effective only for `sig_trace`.
I think the global one should be renamed, on my side I renamed it to `filtered_pid`:

```patch
diff --git a/pkg/gadgets/sigsnoop/tracer/bpf/sigsnoop.bpf.c b/pkg/gadgets/sigsnoop/tracer/bpf/sigsnoop.bpf.c
index eee30636..b3811ae9 100644
--- a/pkg/gadgets/sigsnoop/tracer/bpf/sigsnoop.bpf.c
+++ b/pkg/gadgets/sigsnoop/tracer/bpf/sigsnoop.bpf.c
@@ -7,7 +7,7 @@
 
 #define MAX_ENTRIES    10240
 
-const volatile pid_t target_pid = 0;
+const volatile pid_t filtered_pid = 0;
 const volatile int target_signal = 0;
 const volatile bool failed_only = false;
 const volatile bool filter_by_mnt_ns = false;
@@ -51,7 +51,7 @@ static int probe_entry(pid_t target_pid, int sig)
        pid_tgid = bpf_get_current_pid_tgid();
        pid = pid_tgid >> 32;
        tid = (__u32)pid_tgid;
-       if (target_pid && target_pid != pid)
+       if (filtered_pid && filtered_pid != pid)
                return 0;
 
        event.pid = pid;
@@ -155,7 +155,7 @@ int sig_trace(struct trace_event_raw_signal_generate *ctx)
        pid_tgid = bpf_get_current_pid_tgid();
        pid = pid_tgid >> 32;
        tid = (__u32)pid_tgid;
-       if (target_pid && target_pid != pid)
+       if (filtered_pid && filtered_pid != pid)
                return 0;
 
        event.pid = pid;
```

Best regards.

Reply:
Eagle eye. Thanks. Just updated.